src/lighteval/metrics/llm_as_judge.py

```diff
@@ -96,38 +118,48 @@ def __init__(
         self.url = url
         self.api_key = api_key
         self.backend = judge_backend
+        self.hf_provider = hf_provider
```

Comment:
You need an exception: if backend == inference_provider and hf_provider is None, raise